frontend/test/pytest/test_mid_circuit_measurement.py

```diff
@@ -779,6 +779,26 @@ def cost():
         result = cost()
         assert jnp.array(result).shape == (qubits,)
 
+    def test_dynamic_one_shot_with_classical_return_values(self):
+        """Test shot-vector with complex container sample"""
+        dev = qml.device("lightning.qubit", wires=1, shots=12)
+
+        @qjit
+        @qml.qnode(dev, mcm_method="one-shot")
+        def circuit():
+            qml.Hadamard(0)
+            return {
+                "first": qml.sample(),
+                "second": [100, qml.sample()],
+                "third": (qml.sample(), qml.sample()),
+            }
+
+        assert list(circuit().keys()) == ["first", "second", "third"]
+        assert jnp.array(circuit()["first"]).shape == (12, 1)
+        assert jnp.allclose(circuit()["second"][0], jnp.full(12, 100))
+        assert jnp.array(circuit()["second"][1]).shape == (12, 1)
+        assert jnp.array(circuit()["third"]).shape == (2, 12, 1)
```

Comment:
```suggestion
        result = circuit()
        
        assert list(result.keys()) == ["first", "second", "third"]
        assert jnp.array(result["first"]).shape == (12, 1)
        assert jnp.allclose(result["second"][0], jnp.full(12, 100))
        assert jnp.array(result["second"][1]).shape == (12, 1)
        assert jnp.array(result["third"]).shape == (2, 12, 1)
```
let's save some testing time whenever we can : )

Reply:
Also is it weird that `result["second"][0]` is now an array of 12 `100`s instead of just one `100`? :thinking: What performed the broadcasting?

Reply:
It's not promise that classical values always be the same, taking the following examples, `circuit_1()` and `circuit_2()` would expect to produce `[True|False]+` and `[42|43]+`. But of course, when changing to single-branch, it should only return a single value here.

```python
import pennylane as qml
from catalyst import qjit, measure
@qjit
@qml.set_shots(10)
@qml.qnode(qml.device("lightning.qubit", wires=1), mcm_method="one-shot")
def circuit_1():
    qml.Hadamard(wires=0)
    return measure(0)
print(circuit_1())
```

```python
@qjit(autograph=True)
@qml.set_shots(10)
@qml.qnode(qml.device("lightning.qubit", wires=1), mcm_method="one-shot")
def circuit_2():
    qml.Hadamard(wires=0)
    if measure(0):
        return 42
    else:
        return 43

print(circuit_2())
```

Reply:
Yes the classical values might be different so they need one per shot, but this shouldn't affect dictionary keys (which are never different) since dictionary keys are handled by pytrees. It should still be a pytree with dictionary key 10